### PR TITLE
add support to pre-EIP155 txs

### LINKED ISSUE #1536: Accept legacy transactions

### PR DESCRIPTION
**Pull request by tclemos**
Closes #1536.

### What does this PR do?

add support to pre-EIP155 txs, which are txs signed without Chain ID

### Reviewers

Main reviewers:

@arnaubennassar 
@ToniRamirezM 
@ARR552

### CODE REVIEW

File: state/crypto.go [hunk 1]
```diff
@@ -16,8 +16,10 @@ var (
 func CheckSignature(tx types.Transaction) error {
 	// Check Signature
 	v, r, s := tx.RawSignatureValues()
-	plainV := byte(v.Uint64() - 35 - 2*(tx.ChainId().Uint64()))
-
+	plainV := byte(0)
+	if tx.ChainId().Uint64() != 0 {
```
Comment (cool-develope) [inline state/crypto.go:20]: duplication of ` tx.ChainId().Uint64()`, it is preferred to assign a variable

**Review by ToniRamirezM — APPROVED** [review]

**Review by cool-develope — APPROVED** [review]

File: state/helper.go [hunk 2]
```diff
@@ -12,7 +12,11 @@ import (
 	"github.com/ethereum/go-ethereum/rlp"
 )
 
-const ether155V = 27
+const (
+	double    = 2
+	ether155V = 27
+	etherNewV = 35
```
Comment (ARR552) [inline state/helper.go:18]:
```suggestion
	etherPre155V = 35
```

**Review by ARR552 — APPROVED** [review]
Just a name suggestion

**Review by arnaubennassar — CHANGES_REQUESTED** [review]
It would eb nice to ru an E2E test to ensure that we work well with this kind of txs even when we get them through L1

**Comment by tclemos** [comment]
> It would eb nice to ru an E2E test to ensure that we work well with this kind of txs even when we get them through L1

fair enough, let's do it

**Comment by tclemos** [comment]
> It would eb nice to ru an E2E test to ensure that we work well with this kind of txs even when we get them through L1

@arnaubennassar e2e test added

File: test/e2e/preEIP155_test.go [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+		signedTx, err := types.SignTx(tx, types.HomesteadSigner{}, privateKey)
+		require.NoError(t, err)
+
+		err = client.SendTransaction(ctx, signedTx)
+		require.NoError(t, err)
+
+		err = operations.WaitTxToBeMined(ctx, client, signedTx, operations.DefaultTimeoutTxToBeMined)
+		require.NoError(t, err)
+	}
```
Comment (arnaubennassar) [inline test/e2e/preEIP155_test.go:78]: Could you add wait tx to be virtualized? no need for consolidated, but I want to be sure that we dont have any problem when unpacking the data from the sync

**Review by arnaubennassar — CHANGES_REQUESTED** [review]

**Review by arnaubennassar — APPROVED** [review]

**Pull request merged**